src/applications/auth/containers/AuthApp.jsx

```diff
@@ -133,12 +178,11 @@ export default function AuthApp({ location }) {
 
     if (auth === FORCE_NEEDED) {
       handleAuthForceNeeded();
-    } else if (!hasError && checkReturnUrl(returnUrl)) {
-      handleAuthSuccess({ skipToRedirect: true });
     } else {
       try {
+        const skipToRedirect = !hasError && checkReturnUrl(returnUrl);
         const response = await apiRequest('/user');
-        handleAuthSuccess({ response });
+        await handleAuthSuccess({ response, skipToRedirect });
```

Comment:
Since we need the user profile in the `handleProvisioning` function, we can no longer allow error-free authentications with valid external redirects to skip the `apiRequest('/user')` call.